app/src/main/java/com/example/development/sakaiclient20/networking/deserializers/SiteGradesDeserializer.java

```diff
@@ -0,0 +1,24 @@
+package com.example.development.sakaiclient20.networking.deserializers;
+
+import com.example.development.sakaiclient20.models.sakai.gradebook.SiteGrades;
+import com.example.development.sakaiclient20.networking.deserializers.builders.grades.SiteGradesBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+
+import java.lang.reflect.Type;
+
+/**
+ * Deserializes the Json object into a pojo by delegating to the site grades builder
+ */
+public class SiteGradesDeserializer implements JsonDeserializer<SiteGrades> {
+
+    @Override
+    public SiteGrades deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
+
+        return new SiteGradesBuilder(json.getAsJsonObject())
```

Comment:
I think i told you about this before, but I wasn't able to create a GradeDeserializer because I needed to store the siteID in each grade object, but in the response body, that's only contained in the SiteGrades element, not in each Grade element. So Gson wouldn't know about the siteID if it tried deserializing the grade. So that's why I created the SiteGrades builder so I could manually extract the siteId and then give it to the grade builder for it to store in the grade object.

Reply:
Ah that makes sense. We just need to be extra safe with null and empty checks in that case, but this sounds fine.